syft/pkg/cataloger/golang/scan_binary.go

```diff
@@ -2,46 +2,73 @@ package golang
 
 import (
 	"debug/buildinfo"
+	"debug/elf"
 	"fmt"
 	"io"
+	"io/fs"
+	"regexp"
 	"runtime/debug"
+	"strings"
+	"unicode"
 
-	"github.com/kastenhq/goversion/version"
-
+	"github.com/anchore/syft/internal"
 	"github.com/anchore/syft/internal/log"
 	"github.com/anchore/syft/internal/unknown"
 	"github.com/anchore/syft/syft/file"
 	"github.com/anchore/syft/syft/internal/unionreader"
+	"github.com/kastenhq/goversion/version"
 )
 
 type extendedBuildInfo struct {
 	*debug.BuildInfo
+	sym            []elf.Symbol
 	cryptoSettings []string
 	arch           string
 }
 
+const (
+	devBinaryType  binaryType = "dev"
+	testBinaryType binaryType = "test"
+)
+
+type binaryType string
+
 // scanFile scans file to try to report the Go and module versions.
-func scanFile(location file.Location, reader unionreader.UnionReader) ([]*extendedBuildInfo, error) {
+func (c *goBinaryCataloger) scanFile(location file.Location, reader unionreader.UnionReader) ([]*extendedBuildInfo, binaryType, error) {
 	// NOTE: multiple readers are returned to cover universal binaries, which are files
 	// with more than one binary
 	readers, errs := unionreader.GetReaders(reader)
 	if errs != nil {
 		log.WithFields("error", errs).Debug("failed to open a golang binary")
-		return nil, fmt.Errorf("failed to open a golang binary: %w", errs)
+		return nil, devBinaryType, fmt.Errorf("failed to open a golang binary: %w", errs)
 	}
 
 	var builds []*extendedBuildInfo
+	btyp := devBinaryType
 	for _, r := range readers {
 		bi, err := getBuildInfo(r)
 		if err != nil {
 			log.WithFields("file", location.RealPath, "error", err).Trace("unable to read golang buildinfo")
 
 			continue
 		}
+
 		// it's possible the reader just isn't a go binary, in which case just skip it
 		if bi == nil {
 			continue
 		}
+		var sym []elf.Symbol
+		// a test binary has no dependencies in "buildinfo", so we resort to the .symtab section (i.e. Symbols() )
+		if bi.Deps == nil {
+			sym, err = getSymbolsInfo(r)
+			if err != nil {
+				log.WithFields("file", location.RealPath, "error", err).Trace("unable to read golang symtab")
+				continue
+			}
+			if sym != nil {
+				btyp = testBinaryType
```

Comment:
It looks like the only 2 values returned are `devBinaryType` and `testBinaryType`. It seems these correspond to "dev" and "test" binaries, but what about "normal release" binaries? Release binaries are the primary thing the cataloger is intended to scan, there is probably just a naming thing we should consider changing here